### DISCUSSION TITLE
[2.0-rc update 2.0] About the WebviewWindow.getByLabel method in js

### DISCUSSION THREAD

**Discussion by nongyehong** [tauri-apps/tauri#11351] [Q&A] [answered]
![QQ_1728893396582](https://github.com/user-attachments/assets/9a81370a-2ed3-41e4-8353-e8c1ddafe64f)
From the rc version to the official version, why is this method changed to Promise? I'm a little confused 🤔

**Comment by FabianLars** [comment, marked as answer]
Because tauri's IPC forces the use of async and the function needs to communicate with the rust backend to reliably get the currently available windows ([ref](https://github.com/tauri-apps/tauri/issues/5380)). If this is a problem but you're sure the window exists, you can do this
```ts
new WebviewWindow("your-label", { skip: true })
```
`skip: true` is not part of the public api so typescript will complain but this way it won't try to create a new webview. We used to have a getByLabelUnchecked i think, no idea where it went.

**Reply by nongyehong** [reply]
Ok, I think we need to fix the above example: because there is no Webview method in webviewWindow, newbies may be misled
